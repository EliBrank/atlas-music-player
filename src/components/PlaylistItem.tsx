type PlaylistItemProps = {
  songTitle: string,
  artistName: string,
  songLength: number,
  isSelected: boolean
}

const PlaylistItem = ({songTitle, artistName, songLength, isSelected}: PlaylistItemProps) => {
  const songLengthFormatted: string = getSongLengthFormatted(songLength);
  const bgClass: string = isSelected ? 'bg-selected-light dark:bg-selected-dark' : 'bg-none';

  return (
    <div className={`${bgClass} flex items-center justify-between`}>
      <div>
        <div className="text-base font-medium">
          {songTitle}
        </div>
        <div className="text-sm font-medium text-tertiary-light dark:text-tertiary-dark">
          {artistName}
        </div>
      </div>
      <div className="text-sm font-medium text-tertiary-light dark:text-tertiary-dark">
        {songLengthFormatted}
      </div>
    </div>
  );
}

function getSongLengthFormatted(rawSongDuration: number): string {
  const minutes: number = Math.floor(rawSongDuration / 60);
  const seconds: number = rawSongDuration % 60;
  return `${minutes}:${seconds.toString().padStart(2, '0')}`;
}

export default PlaylistItem;
