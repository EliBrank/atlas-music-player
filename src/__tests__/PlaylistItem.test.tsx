import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlaylistItem from "@/components/PlaylistItem";

test("PlaylistItem renders correctly", () => {
  const onClickTest = () => {return};

  const { container } = render(
    <PlaylistItem
      songTitle="Testing Time"
      artistName="The Testers"
      songLength={220}
      isSelected={true}
      onClick={onClickTest}
    />
  );
  expect(container).toMatchSnapshot();
});
