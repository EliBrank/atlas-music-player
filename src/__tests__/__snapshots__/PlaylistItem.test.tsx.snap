// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlaylistItem renders correctly 1`] = `
<div>
  <div
    class="bg-selected-light dark:bg-selected-dark flex cursor-pointer items-center justify-between"
  >
    <div>
      <div
        class="text-base font-medium"
      >
        Testing Time
      </div>
      <div
        class="text-sm font-medium text-tertiary-light dark:text-tertiary-dark"
      >
        The Testers
      </div>
    </div>
    <div
      class="text-sm font-medium text-tertiary-light dark:text-tertiary-dark"
    >
      3:40
    </div>
  </div>
</div>
`;
